components/scream/cmake/YamlCpp.cmake

```diff
@@ -0,0 +1,34 @@
+# Process the libyaml subdirectory, and make sure it is processed only once
+
+DEFINE_PROPERTY(GLOBAL
+                PROPERTY E3SM_YAMLCPP_ALREADY_BUILT
+                BRIEF_DOCS "Wheter yaml-cpp subdir has already been processed"
+                FULL_DOCS "This property is used by cmake to ensure that yaml-cpp
+                           submodule directory is only included once (with add_subdirectory).")
+
+GET_PROPERTY(IS_YAMLCPP_ALREADY_BUILT GLOBAL PROPERTY E3SM_YAMLCPP_ALREADY_BUILT SET)
+
+IF (NOT IS_YAMLCPP_ALREADY_BUILT)
+  SET (YAMLCPP_SRC_DIR ${SCREAM_SOURCE_DIR}/../../externals/yaml-cpp)
+  SET (YAMLCPP_BIN_DIR ${CMAKE_BINARY_DIR}/externals/yaml-cpp)
+
+  SET (BUILD_TESTING FALSE)
+  ADD_SUBDIRECTORY (${YAMLCPP_SRC_DIR} ${YAMLCPP_BIN_DIR})
```

Comment:
This file is mixing case styles.

Reply:
Do you have any preference on which one to keep? I like upper case for cmake, but we used lower case in other places...

Reply:
all-caps looks awful to me for some reason. I prefer cmake-command to be lowercase, var names upper.